include/exadg/solvers_and_preconditioners/preconditioners/preconditioner_amg.h

```diff
@@ -37,6 +37,44 @@ namespace ExaDG
 {
 using namespace dealii;
 
+template<typename Operator>
+std::unique_ptr<MPI_Comm, void (*)(MPI_Comm *)>
+create_subcommunicator(Operator const & op)
+{
+  unsigned int n_locally_owned_cells = 0;
+  auto const & dof_handler           = op.get_matrix_free().get_dof_handler();
+  for(const auto & cell : dof_handler.active_cell_iterators())
+    if(cell->is_locally_owned())
+      ++n_locally_owned_cells;
+
+  // In case some of the MPI ranks do not have cells, we create a
+  // sub-communicator to exclude all those processes from the PETSc
+  // communication and hence speed up those operations. Note that we have to
+  // free the communicator again, which happens in the preconditioner_amg.h
+  // file which owns the matrix passed to the present function.
+  if(Utilities::MPI::min(n_locally_owned_cells, dof_handler.get_communicator()) == 0)
+  {
+    std::unique_ptr<MPI_Comm, void (*)(MPI_Comm *)> subcommunicator(new MPI_Comm,
+                                                                    [](MPI_Comm * comm) {
+                                                                      MPI_Comm_free(comm);
+                                                                      delete comm;
+                                                                    });
+    MPI_Comm_split(dof_handler.get_communicator(),
+                   n_locally_owned_cells > 0,
+                   Utilities::MPI::this_mpi_process(dof_handler.get_communicator()),
+                   subcommunicator.get());
+    return subcommunicator;
```

Comment:
Please introduce empty lines between the three statements for readability. The same below.